validator/README.md

```diff
@@ -19,30 +19,62 @@ limitations under the License.
 A validator for the
 [AMP HTML format](https://github.com/ampproject/amphtml/blob/master/README.md).
 
+To use, install Node.js version 4.X on your system (tested with v4.4.2). E.g.,
+[by downloading](https://nodejs.org/en/download/) or
+[by using a package manager](https://nodejs.org/en/download/package-manager/) or
+[by using NVM](https://github.com/creationix/nvm).
+
 ## Using the command-line tool
 
-* Install Node.js version 4.X on your system. E.g.,
-  [by downloading](https://nodejs.org/en/download/) or
-  [by using a package manager](https://nodejs.org/en/download/package-manager/) or
-  [by using NVM](https://github.com/creationix/nvm).
-* Type `node index.js` in this directory to get started:
+* Type `./index.js` in this directory to get started:
 
 ```
-$ node index.js testdata/feature_tests/minimum_valid_amp.html
+$ ./index.js testdata/feature_tests/minimum_valid_amp.html
 testdata/feature_tests/minimum_valid_amp.html: PASS
 
-$ node index.js testdata/feature_tests/several_errors.html
+$ ./index.js testdata/feature_tests/several_errors.html
 testdata/feature_tests/several_errors.html:23:2 The attribute 'charset' may not appear in tag 'meta name= and content='.
 testdata/feature_tests/several_errors.html:26:2 The tag 'script' is disallowed except in specific forms.
 testdata/feature_tests/several_errors.html:32:2 The mandatory attribute 'height' is missing in tag 'amp-img'. (see https://www.ampproject.org/docs/reference/amp-img.html)
 testdata/feature_tests/several_errors.html:34:2 The attribute 'width' in tag 'amp-ad' is set to the invalid value '100%'. (see https://www.ampproject.org/docs/reference/amp-ad.html)
 ...
 ```
 
-* If you wish to install the Validator as a system command,
-  install the NPM package manager (e.g. using apt-get in Ubuntu Linux) and
-  run `npm install -g` in this directory. After that, you may type
-  `amp_validator` in any directory to invoke the validator.
+If you wish to install the Validator as a system command,
+install the NPM package manager (e.g. using apt-get in Ubuntu Linux) and
+run `npm install -g` in this directory. After that, you may type
+`amp-validator` in any directory to invoke the validator.
+
+## Using the web UI
+
+This is a simple web editor which validates AMP HTML documents on the fly.
+This feature is new and experimental, feedback is especially welcome.
+
+* Type `./index.js webui` in this directory.
+* Point your web browser to http://127.0.0.1:8765/
+
+## Using the NodeJS API
+
+This API is new and still experimental, feedback is especially welcome. We may
+try to port to earlier versions of NodeJS if sufficient interest exists.
+
+```
+'use strict';
+
+const ampValidator = require('amp-validator');
+ampValidator.validateString('<html>Hello, world</html>').then(
+    (result) => {
+        ((result.status === 'PASS') ? console.log : console.error)(result.status);
```

Comment:
> I think this would have the problem that the promise could either be rejected due to a problem instantiating the validator or something wrong with the page.

I think this is going to be inherent since we're not bundling the validator inside `amp-validator` (why is that?)

> But validateString would just return a ValidationResult object, which neatly corresponds to the proto buffer

That's better, but still odd that I can't just tell my `ampValidator` instance to validate.